sprokit/processes/core/write_object_track_process.cxx

```diff
@@ -133,21 +134,29 @@ ::_init()
 void write_object_track_process
 ::_step()
 {
-  std::string file_name;
+  auto const& port_info = peek_at_port_using_trait( object_track_set );
+  if( port_info.datum->type() == sprokit::datum::complete )
+  {
+    grab_edge_datum_using_trait( object_track_set );
```

Comment:
I would also grab the timestamp now since you made it required so it's not sitting on the edge

Reply:
That port is now optional.

Reply:
IIUC, yes. @mattdawkins, who wrote this bit, can give a longer answer.

Reply:
Yes, some writers don't write out their tracks until the very end. This is also due to the fact that some python implementations and derived classes don't get their destructors called properly so after the last frame but before the desctructor is called we wanted to write out all tracks.